libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -243,3 +288,46 @@ export function processMartechAttributeMetadata(html) {
   });
   return null;
 }
+
+export async function fetchAndProcess({ url, message, shouldDecorateLinks = true } = {}) {
+  try {
+    const path = getFederatedUrl(url);
+    const res = await fetch(path.replace(/(\.html$|$)/, '.plain.html'));
+    const text = await res.text();
+    const { body } = new DOMParser().parseFromString(text, 'text/html');
+
+    const inlineFrags = [...body.querySelectorAll('a[href*="#_inline"]')];
+    if (inlineFrags.length) {
+      const { default: loadInlineFrags } = await import('../../fragment/fragment.js');
+
+      const fragPromises = inlineFrags.map((link) => {
+        // Replacing paragraphs should happen in the fragment module
+        // https://jira.corp.adobe.com/browse/MWPW-141039
+        if (link.parentElement && link.parentElement.nodeName === 'P') {
+          const div = document.createElement('div');
+          link.parentElement.replaceWith(div);
+          div.appendChild(link);
+        }
+        link.href = getFederatedUrl(link.href);
+        return loadInlineFrags(link);
+      });
+      await Promise.all(fragPromises);
+    }
+
+    if (path.includes('/federal/')) federatePictureSources(body);
+
+    if (shouldDecorateLinks) decorateLinks(body);
+
+    const blocks = body.querySelectorAll('.martech-metadata');
+    blocks.forEach((block) => {
+      import('../../martech-metadata/martech-metadata.js')
+        .then(({ default: decorate }) => decorate(block));
+    });
+
+    body.innerHTML = await replaceText(body.innerHTML, getFedsPlaceholderConfig());
+    return body;
+  } catch (e) {
+    lanaLog({ message: `${message} url: ${url}`, e, tags: 'errorType=error,module=utilities' });
```

Comment:
I can't say I'm entirely sold on having the message here. Could we not tie it to where we actually call the method and avoid passing it in as a param?

Reply:
I'm a bit cautious as the method carries a lot of logic and can error.  So we'd always need a try/catch upstream of this method if we removed the try/catch here. If we'd just remove the error message itself and keep the try catch, we'd lose context of _what_ errored.

Reply:
I'l all for keeping the `try/catch` here, just maybe throw an error and handle it in the module itself, so that the error message is closer to the logic itself; might also make it easier to investigate LANA logs

Reply:
Slightly changed the logic to bring the lana logs closer to where they should be